src/fmu/tools/qcforward/_qcforward.py

```diff
@@ -197,3 +210,47 @@ def parse(self, rule):
         self.expression = key + self.compare + str(self.limit) + "%"
         if self.given:
             self.expression += "ifx" + self.given + str(self.criteria)
+
+
+def actions_validator(actionsin: dict) -> dict:
+    """General function to validate that the 'actions' input is on the correct form.
+
+    The typical form is::
+
+        "actions": [
+            {"warn": "anywell < 80%", "stop": "anywell < 75%"},
+            {"warn": "allwells < 90%", "stop": "allwells < 85%"},
+        ],
+
+    Several checks can be implemented; currently many is in a to-do state.
+
+    Args:
+        actionsin: Input actions record
+
+    Returns:
+        An actions python dictionary (potentially processed)
+    """
+
+    actions = deepcopy(actionsin)
+
+    if not isinstance(actions, list):
+        raise ValueError("The actions input must be a list")
+
+    if len(actions) != 2:
+        raise ValueError("The actions input must be a list with two element")
+
+    # check each line to see if the rules seems logical
+    for elem in actions:
+        if not set(elem.keys()).issubset(["warn", "stop"]):
+            raise ValueError(
+                "Both criteria 'warn' and 'stop' are required in actions! "
+                f"You have keys: {list(elem.keys())}"
+            )
+        vals = list(elem.values())
+        if "any" in str(vals) and "all" in str(vals):
+            # avoid that one entry mixes all and any
+            raise ValueError(
+                "Seems wrong to mix 'all' and 'any' in one line; be consistent"
+            )
+
```

Comment:
Should also include an Error if the action does not contain `all` or `any` to avoid error later

Reply:
Possibly